core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -85,7 +85,8 @@ private[spark] class CoarseGrainedExecutorBackend(
 
     logInfo("Connecting to driver: " + driverUrl)
     try {
-      if (PlatformDependent.directBufferPreferred() &&
+      if (env.conf.getBoolean("spark.shuffle.io.preferDirectBufs", true) &&
```

Comment:
Should this check be 'and' or 'or'? how is it implemented elsewhere? that is, should 'true' or 'false' setting of this value override the default?

Reply:
The configuration key and default value come from `TransportConf#preferDirectBufs`

Reply:
OK so the setting is really only a way to disable, not enable, direct buffers